tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/RapidsJsonConfTrait.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "330"}
+spark-rapids-shim-json-lines ***/
+package org.apache.spark.sql.rapids.utils
+
+import org.scalatest.{BeforeAndAfterAll, Suite}
+
+import org.apache.spark.sql.internal.SQLConf
+
+trait RapidsJsonConfTrait extends BeforeAndAfterAll { this: Suite =>
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    SQLConf.get.setConfString("spark.rapids.sql.expression.JsonTuple", true.toString)
+    SQLConf.get.setConfString("spark.rapids.sql.expression.GetJsonObject", true.toString)
+    SQLConf.get.setConfString("spark.rapids.sql.expression.JsonToStructs", true.toString)
+    SQLConf.get.setConfString("spark.rapids.sql.expression.StructsToJson", true.toString)
+  }
+
+  override def afterAll(): Unit = {
+    SQLConf.get.unsetConf("spark.rapids.sql.expression.JsonTuple")
+    SQLConf.get.unsetConf("spark.rapids.sql.expression.GetJsonObject")
+    SQLConf.get.unsetConf("spark.rapids.sql.expression.JsonToStructs")
+    SQLConf.get.unsetConf("spark.rapids.sql.expression.StructsToJson")
+    super.afterAll()
+  }
```

Comment:
It is not how I thought my previous comment would be addressed. Should not beforeAll and afterAll largely be the same implementation as in 
other Unit Tests  based off SparkQueryCompareTestSuite

especially the cleanup after?

```
 override def afterAll(): Unit = { 
   super.afterAll() 
   TrampolineUtil.cleanupAnyExistingSession() 
 } 
```

And we should just make sure that we reuse for new UT and

Reply:
RapidsJsonConfTrait is an extra trait to inject more json related configs, and it will call the beforeAll/afterAll methods in its parent tratis ( because of trait linerlization)

Reply:
I am talking about the fact that we already have the code cleaning out SparkSession via `TrampolineUtil.cleanupAnyExistingSession()` . Instead of doing a special logic for this one test we could reuse the more general cleanup code.

Reply:
As discussed with Gera offline, we'll use https://github.com/NVIDIA/spark-rapids/issues/10886 to track this